x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/prebuilt_rules/prebuilt_rules_preview.cy.ts

```diff
@@ -1049,5 +1062,104 @@ describe('Detection rules, Prebuilt Rules Installation and Update workflow', ()
         }
       );
     });
+
+    describe('Viewing rule changes in JSON diff view', { tags: TEST_ENV_TAGS }, () => {
+      it('User can see changes in a side-by-side JSON diff view', () => {
+        clickRuleUpdatesTab();
+
+        openRuleUpdatePreview(OUTDATED_RULE_1['security-rule'].name);
+        assertSelectedPreviewTab('JSON view');
+
+        cy.get(UPDATE_PREBUILT_RULE_PREVIEW)
+          .contains('Current rule')
+          .as('currentRuleColumnHeading');
+        cy.get(UPDATE_PREBUILT_RULE_PREVIEW)
+          .contains('Elastic update')
+          .as('elasticUpdateColumnHeading');
+
+        cy.get('@currentRuleColumnHeading').should('be.visible');
+        cy.get('@elasticUpdateColumnHeading').should('be.visible');
+
+        /* "Current rule" should be the left column, "Elastic update" should be the right column. */
+        cy.get('@currentRuleColumnHeading').then(($currentRuleColumnHeading) => {
+          const currentRuleLeftOffset = $currentRuleColumnHeading.offset()?.left || 0;
+          cy.get('@elasticUpdateColumnHeading').then(($elasticUpdateColumnHeading) => {
+            const elasticUpdateLeftOffset = $elasticUpdateColumnHeading.offset()?.left || 0;
+            expect(currentRuleLeftOffset).to.be.lessThan(elasticUpdateLeftOffset);
+          });
+        });
+
+        cy.get(UPDATE_PREBUILT_RULE_PREVIEW).contains('"version": 1').as('currentRuleValue');
+        cy.get(UPDATE_PREBUILT_RULE_PREVIEW).contains('"version": 2').as('updateVersionValue');
+
+        cy.get('@currentRuleValue').should('be.visible');
+        cy.get('@updateVersionValue').should('be.visible');
+
+        /*
+          "Current rule" version value should be displayed in the left column, 
+          "Elastic update" version value should be displayed the right column.
+        */
+        cy.get('@currentRuleValue').then(($currentRuleValue) => {
+          const currentRuleLeftOffset = $currentRuleValue.offset()?.left || 0;
+          cy.get('@updateVersionValue').then(($updateVersionValue) => {
+            const updateVersionLeftOffset = $updateVersionValue.offset()?.left || 0;
+            expect(currentRuleLeftOffset).to.be.lessThan(updateVersionLeftOffset);
+          });
+        });
+
+        cy.get(UPDATE_PREBUILT_RULE_PREVIEW)
+          .contains('"name": "Outdated rule 1"')
+          .should('be.visible');
+
+        /* Select another rule without closing the preview for the current rule */
+        openRuleUpdatePreview(OUTDATED_RULE_2['security-rule'].name);
+
+        /* Make sure the JSON diff is displayed for the newly selected rule */
+        cy.get(UPDATE_PREBUILT_RULE_PREVIEW)
+          .contains('"name": "Outdated rule 2"')
+          .should('be.visible');
+        cy.get(UPDATE_PREBUILT_RULE_PREVIEW)
+          .contains('"name": "Outdated rule 1"')
+          .should('not.exist');
+      });
+
+      it('Dynamic properties should not be included in preview', () => {
```

Comment:
This test only checks for `execution_summary`. What about the other fields specified in the test plan like `revision` and `updated_at`?

Reply:
These are in the React Testing Library test called "Technical properties should not be included in preview". I separated "dynamic" from "technical" because I wanted to test the "dynamic" ones in a more realistic way with a Cypress E2E test.